BuildResidentialScheduleFile/resources/schedules.rb

```diff
@@ -1415,20 +1408,29 @@ def generate_cooking_power_schedule(mkc_activity_schedules)
     return cooking_power_sch
   end
 
-  # Apply random time shift and normalize schedule values.
+  # Apply random time shift to schedule values without normalizing.
   #
   # @param schedule [Array<Float>] Array of minute-level schedule values
-  # @param random_offset [Integer] Random offset in minutes to apply to the schedule
-  # @return [Array<Float>] Normalized schedule with random time shift applied
-  def random_shift_and_normalize(schedule, random_offset)
-    schedule = schedule.rotate(random_offset)
+  # @return [Array<Float>] Schedule with random time shift applied
+  def random_shift_and_aggregate(schedule)
+    schedule = schedule.rotate(@random_offset)
```

Comment:
I assume we can prevent making a copy here by using 

```suggestion
    schedule.rotate!(@random_offset)
```

Reply:
Done. It took a lot more than this (this particular fix didn't register much change in timing) but think still worthwhile.